packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -91,16 +105,45 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
+        const { paymentData } = payment;
+        const isVaultedFlow =
+            paymentData && isVaultedInstrument(paymentData) && isHostedInstrumentLike(paymentData);
+
         if (!this.orderId) {
-            throw new PaymentMethodInvalidError();
+            if (isVaultedFlow) {
+                this.orderId = await this.createOrder();
+            } else {
```

Comment:
Should we throw an error if `this.createOrder()` throws an error or does not return valid `orderId`?